### DISCUSSION TITLE
How does a backup and restore process go?

### DISCUSSION THREAD

**Discussion by Bubbelb** [nextcloud/all-in-one#4969] [Questions] [answered]
Hi,
I'm planning to implement my own backup solution, instead of using the Nextcloud AIO solution. I know about the start/stop procedures of the whole Nextcloud-AIO system via command line, but what i cannot find is the following:

* Which volumes need to be backed-up for a complete backup (including community containers, like Valutwarden, Caddy, etc.)?
*  What is the procedure to manually restore an AIO instance from scratch? I assume something like this:
  * Recreate and restore all backed-up volumes
  * Start the AIO Master container with the same config/parameter set as the original.
  * Restart all the containers and done... I Guess.

Are there other things to consider?

Thanks in advance.

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by Bubbelb** [reply]
Thank you. As I read it again (I already read it earlier, but didn't realise it almost answered my question). I am not totally sure it's complete, though. So, for the sake of the discussion, I made a backup via the AIO tooling, and compared it to all the nextlcoud_aio volumes I have:

| Name | As Docker Volume |  in AIO Backup | Notes |
| --- | --- | --- | --- |
| nextcloud_aio_apache | Y | Y | Don't know why this should be backed-up. |
| nextcloud_aio_backup_cache  | Y | N | Part of AIO Borg Backup. Logical not backup. |
| nextcloud_aio_caddy | Y | N | Only an application, no significant data t backup. |
| nextcloud_aio_database  | Y | Y | Database files at rest, because SQL is not active during backup. |
| nextcloud_aio_database_dump | Y | Y | Database backups (Exports so to speak). How are those created? Are they needed? |
| nextcloud_aio_elasticsearch | Y | Y | Handier to backup search indexes, than to recreate them, I guess. |
| nextcloud_aio_mastercontainer | Y | Y | Don't know the exact content. |
| nextcloud_aio_nextcloud | Y | Y | The config files maybe of Nextcloud? |
| nextcloud_aio_nextcloud_data | Y | Y | The actual data. |
| nextcloud_aio_redis | Y | Y | The Redis Cache. Why is this needed in a backup? |
| nextcloud_aio_vaultwarden | Y | Y | My Vaultwarden vaults |
| nextcloud_aio_vaultwarden_logs | Y | N | Logs, not important, so it seems. |

I know this is not complete, because I have no Jellyfin, Plex or LDAP running inside Nextcloud_AIO, but it's a start. Maybe the above table is something for in the Wiki, I don't know.

But my point is: Do I need to recreate the volumes myself, upon restore, and if so, does that also count for the volumes not backed-up, like vautwarden_logs, for example.

And is my assumption correct for a restore:
* Recreate Backup volumes manually and restore the data in them.
* Restart mastercontainer with the same parameters as at the time of the backup.

And, out of curiosity: What is the database_dump for? The database files are already backed up in a consistent manner, because the database engine is stopped before the backup is started.

Probablu a far too long story. :) Sorry for that.
